src/palace/manager/celery/tasks/boundless.py

```diff
@@ -1,520 +1,112 @@
-from sqlalchemy.orm.exc import ObjectDeletedError, StaleDataError
+from celery.canvas import Signature
 
-from palace.manager.api.circulation.base import (
-    BaseCirculationAPI,
-    LibrarySettingsType,
-    SettingsType,
+from palace.manager.celery.importer import (
+    import_all as create_import_tasks,
+    import_key,
+    import_lock,
 )
 from palace.manager.celery.task import Task
+from palace.manager.celery.tasks import apply, identifiers
 from palace.manager.celery.utils import load_from_id
 from palace.manager.core.exceptions import IntegrationException
-from palace.manager.data_layer.bibliographic import BibliographicData
 from palace.manager.integration.license.boundless.api import BoundlessApi
-from palace.manager.integration.license.boundless.requests import BoundlessRequests
+from palace.manager.integration.license.boundless.importer import BoundlessImporter
 from palace.manager.service.celery.celery import QueueNames
-from palace.manager.service.redis.models.lock import RedisLock
-from palace.manager.service.redis.redis import Redis
+from palace.manager.service.redis.models.set import IdentifierSet
 from palace.manager.sqlalchemy.model.collection import Collection
-from palace.manager.sqlalchemy.model.coverage import Timestamp
-from palace.manager.sqlalchemy.model.identifier import Identifier
-from palace.manager.sqlalchemy.model.licensing import LicensePool
-from palace.manager.sqlalchemy.util import get_one_or_create
-from palace.manager.util.backoff import exponential_backoff
-from palace.manager.util.datetime_helpers import datetime_utc, utc_now
-from palace.manager.util.http import BadResponseException
-from palace.manager.util.log import pluralize
-
-DEFAULT_BATCH_SIZE: int = 50
-DEFAULT_START_TIME = datetime_utc(1970, 1, 1)
-MAX_DELAY_IN_SECONDS_BETWEEN_REAP_TASKS = 20
 
 
 @shared_task(queue=QueueNames.default, bind=True)
-def import_all_collections(
-    task: Task, import_all: bool = False, batch_size: int = DEFAULT_BATCH_SIZE
-) -> None:
+def import_all_collections(task: Task, *, import_all: bool = False) -> None:
     """
     A shared task that loops through all Boundless Api based collections and kick off an
     import task for each.
     """
     with task.session() as session:
-        count = 0
-        for collection in get_collections_by_protocol(
-            task=task, session=session, protocol_class=BoundlessApi
-        ):
-            task.log.info(
-                f'Queued collection("{collection.name}" [id={collection.id}] for importing...'
-            )
-            list_identifiers_for_import.apply_async(
-                kwargs={"collection_id": collection.id},
-                countdown=count * 5,  # stagger the execution of the collection import
-                # in order to minimize chance of deadlocks caused by
-                # simultaneous updates to the metadata when CM has multiple
-                # Boundless collections configured with overlapping content
```

Comment:
I'm hoping this won't be necessary now, since the apply task is protected by a lock, and retries if it can't get the lock. Overlapping tasks should just wait their turn in the queue.